autorest/validations.go

```diff
@@ -0,0 +1,359 @@
+package autorest
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+// Constraint stroes constraint name, target field name
+// Rule and chain validations.
+type Constraint struct {
+
+	// Target Field name for validation.
+	Target string
+
+	// Constraint name .e.g. Minlength, Maxlength, Pattern, etc.
+	Name string
+
+	// Rule for constraint .e.g. greater than 10, less than 5 etc.
+	Rule interface{}
+
+	// Chain Validations for struct type
+	Chain *[]Constraint
+}
+
+// Validation stores parameter-wise validation.
+type Validation struct {
+	TargetValue interface{}
+	Constraints []Constraint
+}
+
+const (
+	empty            = "Empty"
+	null             = "Null"
+	readOnly         = "ReadOnly"
+	pattern          = "Pattern"
+	maxLength        = "MaxLength"
+	minLength        = "MinLength"
+	maxItems         = "MaxItems"
+	minItems         = "MinItems"
+	multipleOf       = "MultipleOf"
+	uniqueItems      = "UniqueItems"
+	inclusiveMaximum = "InclusiveMaximum"
```

Comment:
can't we just export these constants as integers (with iota)? Do they have to have string values?